src/server.c

```diff
@@ -6970,16 +6971,20 @@ int main(int argc, char **argv) {
             /* All the other options are parsed and conceptually appended to the
              * configuration file. For instance --port 6380 will generate the
              * string "port 6380\n" to be parsed after the actual config file
-             * and stdin input are parsed (if they exist). */
-            else if (argv[j][0] == '-' && argv[j][1] == '-') {
+             * and stdin input are parsed (if they exist).
+	     * Only consider that if the last config has at least one argument. */
+            else if (last_config_args && argv[j][0] == '-' && argv[j][1] == '-') {
```

Comment:
yes, indeed. updated the top comment

Reply:
@yoav-steinberg i'm not sure i understand what exactly would break here.
the only possible issue with this change could be if there's some config that could have handled 0 arguments, and people relied on it.
like doing `src/redis-server --save --loglevel verbose`
i.e. assuming that not passing any value to the save config, would work.

Reply:
actually, testing it now, i see it used to work. 8-(
so this could possibly be a breaking change that we can't afford to introduce now.

are there any other multi-arg configs that are happy with 0 args?

Reply:
silly question (was already answered).
`ALLOW_EMPTY_ARG_CONFIG` was applied to `bind`, `save` and `latency-tracking-info-percentiles`

Reply:
@madolson @yossigo note the breaking change above.
`src/redis-server --save --loglevel verbose` used to work the same as `src/redis-server --save "" --loglevel verbose`
maybe we can't afford to introduce this change now that 7.0 is out, and we need to revert the second change of the change this PR brings (the one in server.c).

doing so will still allow users to pass multi-args configs as a single argv item, just not use `--` as a config value.

Reply:
not worth breaking `src/redis-server --save --loglevel verbose`, i revert it
also I keep these tests in case they are broken in the future (it may feel a little redundant)